src/translation-service.js

```diff
@@ -7,37 +7,37 @@
 
 /**
  * @module utils/translation-service
- */
-
-let dictionaries = {};
-
-/**
- * Adds package translations to existing ones.
- * These translations will later be available for {@link module:utils/translation-service~translate translate}.
  *
- *		add( 'pl', {
- *			'OK': 'OK',
- *			'Cancel [context: reject]': 'Anuluj'
- *		} );
- *
- * That function is accessible globally via `window.CKEDITOR_TRANSLATIONS.add()`. So it's possible to add translation from
- * the other script, just after that one.
+ * Translation service provides {module:utils/translation-service.translate translate} method which can be used
+ * to translate phrase to the given language. Translation should be previously added directly to
+ * the window.CKEDITOR_TRANSLATIONS variable, safely extending current ones.
  *
  * 		<script src="./path/to/ckeditor.js"></script>
- * 		<script src="./path/to/translations/en.js"></script>
+ * 		<script src="./path/to/translations/de.js"></script>
  *
- * @param {String} lang Target language.
- * @param {Object.<String, String>} translations Translations which will be added to the dictionary.
+ * Example of the function that can add translations to the given language.
+ *
+ * 		function addTranslations( lang, translations ) {
+ *			if ( !window.CKEDITOR_TRANSLATIONS ) {
+ *				window.CKEDITOR_TRANSLATIONS = {};
+ *			}
+ *
+ *			const dictionary = window.CKEDITOR_TRANSLATIONS[ lang ] || ( window.CKEDITOR_TRANSLATIONS[ lang ] = {} );
+ *
+ *			// Extend the dictionary for the given language.
+ *			Object.assign( dictionary, translations );
+ *		}
  */
-export function add( lang, translations ) {
-	dictionaries[ lang ] = dictionaries[ lang ] || {};
 
-	Object.assign( dictionaries[ lang ], translations );
+// Initialize CKEDITOR_TRANSLATIONS if it's not initialized.
+if ( !window.CKEDITOR_TRANSLATIONS ) {
```

Comment:
Again, I think that such an access to the global variable is unsafe in envs like Electron, but let's go with it for now. I'll only change it so it imports the global object from `src/global.js`.

Reply:
Or maybe I won't... because it's `src/dom/global.js`. I'll report a followup.